tests/everflow/test_everflow_testbed.py

```diff
@@ -788,6 +788,84 @@ def background_traffic(run_count=None):
+            if mirror_session["session_name"]:
+                remove_command = "config mirror_session remove {}".format(mirror_session["session_name"])
+                for duthost in duthost_set:
+                    duthost.command(remove_command)
+                add_command = "config mirror_session add {} {} {} {} {} {} {}" \
+                    .format(mirror_session["session_name"],
+                            mirror_session["session_src_ip"],
+                            mirror_session["session_dst_ip"],
+                            mirror_session["session_dscp"],
+                            mirror_session["session_ttl"],
+                            mirror_session["session_gre"],
+                            queue_num)
+                for duthost in duthost_set:
+                    duthost.command(add_command)
+            else:
+                pytest.skip("Mirror session info is empty, can't proceed further!")
+
+        queue = str(random.randint(1, 7))
+        # Apply mirror session config with a different queue value other than default '0'
+        configure_mirror_session_with_queue(setup_mirror_session, queue)
+
+        # Add a route to the mirror session destination IP
+        tx_port = setup_info[dest_port_type]["dest_port"][0]
+        peer_ip = everflow_utils.get_neighbor_info(remote_dut, tx_port, tbinfo)
+        everflow_utils.add_route(remote_dut, setup_mirror_session["session_prefixes"][0], peer_ip,
+                                 setup_info[dest_port_type]["remote_namespace"])
+
+        time.sleep(15)
+
+        # Verify that mirrored traffic is sent along the route we installed
+        rx_port_ptf_id = setup_info[dest_port_type]["src_port_ptf_id"]
+        tx_port_ptf_id = setup_info[dest_port_type]["dest_port_ptf_id"][0]
+        src_port = setup_info[dest_port_type]["src_port"]
+        # Clear queue counters for the port asic instance
+        asic_ns = everflow_dut.get_port_asic_instance(src_port).get_asic_namespace()
+        asic_id = everflow_dut.get_asic_id_from_namespace(asic_ns)
+        everflow_dut_asichost = everflow_dut.asic_instance(asic_id)
+        everflow_utils.clear_queue_counters(everflow_dut_asichost)
+        recircle_port = "Ethernet-Rec{}".format(asic_id)
+
+        self._run_everflow_test_scenarios(
+            ptfadapter,
+            setup_info,
+            setup_mirror_session,
+            everflow_dut,
+            rx_port_ptf_id,
+            [tx_port_ptf_id],
+            dest_port_type
+        )
+        # Assert the specific asic recircle port's queue
+        # Make sure mirrored packets are sent via specific queue configured
+        for q in range(1, 8):
+            if str(q) == queue:
+                out = wait_until(30, 1, 0, everflow_utils.check_queue_counters, everflow_dut, asic_ns, recircle_port, q)
+                assert out is True, 'Recircle port {} queue counter {} value is 0'.format(recircle_port, q)
```

Comment:
can we check if we send packet 100 counter should be 100 ?

Reply:
Thanks, Abhisek, for reviewing. Added code to check packet counts in iterations of 10, 50, 100 and verified the same.

![image](https://github.com/user-attachments/assets/ef7c4346-422d-481c-a586-b1e9d0fea319)